File: BeispielProjekt/JDomain/src/main/java/de/beispiel/BeispielDelegate.java
package de.beispiel;

/**
 * Delegate von Beispiel
 * Wird nur initial generiert.
 * Methoden müssen implementiert werden.
 **/
public class BeispielDelegate {

    public String printBeispiel(Beispiel caller){
        return "Beispiel: \n" + caller.getInhalt();
    }

    public String printBeispielMarkdown(Beispiel caller){
        return "";
    }
}

File: BeispielProjekt/TSProjekt/src/delegates/de/beispiel/BeispielDelegate.ts
import {Beispiel} from "../../../de/beispiel/Beispiel";
import {BeispielTyp} from "../../../de/beispiel/BeispielTyp";


export function printBeispiel(caller: Beispiel): string {
    return `${caller.typ == BeispielTyp.TEXT ? "Text-" : "Code-"}Beispiel ${caller.i}`;
}

export function printBeispielMarkdown(caller: Beispiel): string {
    return `## Beispiel\n${caller.typ == BeispielTyp.TEXT ? "" : "```javascript\n"}${caller.inhalt}${caller.typ == BeispielTyp.TEXT ? "" : "\n```"}`;
}
